include/nbl/builtin/hlsl/rwmc/resolve.hlsl

```diff
@@ -0,0 +1,191 @@
+#ifndef _NBL_BUILTIN_HLSL_RWMC_RESOLVE_HLSL_INCLUDED_
+#define _NBL_BUILTIN_HLSL_RWMC_RESOLVE_HLSL_INCLUDED_
+
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+#include <nbl/builtin/hlsl/colorspace/encodeCIEXYZ.hlsl>
+#include <nbl/builtin/hlsl/rwmc/ResolveParameters.hlsl>
+#include <nbl/builtin/hlsl/concepts/accessors/loadable_image.hlsl>
+#include <nbl/builtin/hlsl/colorspace.hlsl>
+#include <nbl/builtin/hlsl/vector_utils/vector_traits.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace rwmc
+{
+		// declare concept
+#define NBL_CONCEPT_NAME ResolveAccessorBase
+#define NBL_CONCEPT_TPLT_PRM_KINDS (typename)(typename)(int32_t)
+#define NBL_CONCEPT_TPLT_PRM_NAMES (T)(VectorScalarType)(Dims)
+// not the greatest syntax but works
+#define NBL_CONCEPT_PARAM_0 (a,T)
+#define NBL_CONCEPT_PARAM_1 (scalar,VectorScalarType)
+// start concept
+	NBL_CONCEPT_BEGIN(2)
+// need to be defined AFTER the concept begins
+#define a NBL_CONCEPT_PARAM_T NBL_CONCEPT_PARAM_0
+#define scalar NBL_CONCEPT_PARAM_T NBL_CONCEPT_PARAM_1
+NBL_CONCEPT_END(
+	((NBL_CONCEPT_REQ_EXPR)((a.calcLuma(vector<VectorScalarType, 3>(scalar, scalar, scalar)))))
+);
+#undef a
+#undef scalar
+#include <nbl/builtin/hlsl/concepts/__end.hlsl>
+
+/* ResolveAccessor is required to:
+*	- satisfy `LoadableImage` concept requirements
+*	- implement function called `calcLuma` which calculates luma from a 3 component pixel value
+*/
+
+template<typename T, typename VectorScalarType, int32_t Dims>
+NBL_BOOL_CONCEPT ResolveAccessor = ResolveAccessorBase<T, VectorScalarType, Dims> && concepts::accessors::LoadableImage<T, VectorScalarType, Dims>;
+
+template<typename OutputScalar>
+struct ResolveAccessorAdaptor
+{
+	using output_scalar_type = OutputScalar;
+	using output_type = vector<OutputScalar, 4>;
+	NBL_CONSTEXPR int32_t image_dimension = 2;
+
+	RWTexture2DArray<float32_t4> cascade;

```

Comment:
This needs to be address right away https://github.com/Devsh-Graphics-Programming/Nabla/pull/959#discussion_r2589172822